lib/ufe/UsdTransform3d.cpp

```diff
@@ -91,78 +122,65 @@ Ufe::SceneItem::Ptr UsdTransform3d::sceneItem() const
 
 Ufe::TranslateUndoableCommand::Ptr UsdTransform3d::translateCmd()
 {
-	auto translateCmd = UsdTranslateUndoableCommand::create(fPrim, fItem->path(), fItem);
+	auto translateCmd = UsdTranslateUndoableCommand::create(fPrim, fItem->path(), fItem, timeCode());
 	return translateCmd;
 }
 
 void UsdTransform3d::translate(double x, double y, double z)
 {
-	translateOp(fPrim, fItem->path(), x, y, z);
+	MayaUsdUtils::MayaTransformAPI api(fPrim);
+	api.translate(GfVec3f(x, y, z), timeCode());
 }
 
 Ufe::Vector3d UsdTransform3d::translation() const
 {
-	double x{0}, y{0}, z{0};
-	const TfToken xlate("xformOp:translate");
-	if (fPrim.HasAttribute(xlate))
-	{
-		// Initially, attribute can be created, but have no value.
-		GfVec3d v;
-		if (fPrim.GetAttribute(xlate).Get<GfVec3d>(&v))
-		{
-			x = v[0]; y = v[1]; z = v[2];
-		}
-	}
-	return Ufe::Vector3d(x, y, z);
+	MayaUsdUtils::MayaTransformAPI api(fPrim);
+	auto translate = api.translate(timeCode());
+	return Ufe::Vector3d(translate[0], translate[1], translate[2]);
 }
 
 Ufe::RotateUndoableCommand::Ptr UsdTransform3d::rotateCmd()
 {
-	auto rotateCmd = UsdRotateUndoableCommand::create(fPrim, fItem->path(), fItem);
+	auto rotateCmd = UsdRotateUndoableCommand::create(fPrim, fItem->path(), fItem, timeCode());
```

Comment:
I see at least three separate changes/PRs in here. 
1) the addition of new TransformAPI (this is already a separate PR)
2) enabling usage of new TransformAPI (can include new arguments to pass time, but shouldn't modify the behavior yet)
3) switching to set data at a `current time`

Points 2 and 3 require separate discussions.